File: src/main/java/org/goodgoodgood/freetubeexservice/dto/music/PlaylistsShortDto.java
package org.goodgoodgood.freetubeexservice.dto.music;

import lombok.Getter;
import org.goodgoodgood.freetubeexservice.domain.music.Playlist;

@Getter
public class PlaylistsShortDto {

    private Long id;
    private String title;

    public PlaylistsShortDto(Playlist playlist) {
        this.id = playlist.getId();
        this.title = playlist.getTitle();
    }
}

File: src/main/kotlin/org/goodgoodgood/freetubeexservice/domain/music/dto/PlaylistsShortDto.kt
package org.goodgoodgood.freetubeexservice.domain.music.dto

import org.goodgoodgood.freetubeexservice.domain.music.Playlist

data class PlaylistsShortDto(
    val id: Long,
    val title: String,
) {

    companion object {
        fun of(playlist: Playlist): PlaylistsShortDto {
            return PlaylistsShortDto(
                id = playlist.id!!,
                title = playlist.title,
            )
        }
    }
}